ui/apps/platform/src/Containers/Collections/CollectionsTable.tsx

```diff
@@ -1,24 +1,54 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { useHistory } from 'react-router-dom';
-import { Button, ButtonVariant, Truncate, Bullseye, Text } from '@patternfly/react-core';
+import {
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+    Pagination,
+    Button,
+    ButtonVariant,
+    Truncate,
+    SearchInput,
+    Bullseye,
+    Text,
+} from '@patternfly/react-core';
 import { SearchIcon } from '@patternfly/react-icons';
 import { TableComposable, TableVariant, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import debounce from 'lodash/debounce';
 
 import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate';
 import LinkShim from 'Components/PatternFly/LinkShim';
 import useTableSelection from 'hooks/useTableSelection';
-import { CollectionResponse } from 'services/CollectionsService';
+import { UseURLPaginationResult } from 'hooks/useURLPagination';
 import { GetSortParams } from 'hooks/useURLSort';
+import { CollectionResponse } from 'services/CollectionsService';
+import { SearchFilter } from 'types/search';
 import { collectionsPath } from 'routePaths';
 
 export type CollectionsTableProps = {
     collections: CollectionResponse[];
+    collectionsCount: number;
+    pagination: UseURLPaginationResult;
+    searchFilter: SearchFilter;
+    setSearchFilter: (searchFilter: SearchFilter) => void;
     getSortParams: GetSortParams;
     hasWriteAccess: boolean;
 };
 
-function CollectionsTable({ collections, getSortParams, hasWriteAccess }: CollectionsTableProps) {
+// TODO We are using this value elsewhere in the app - extract to a central location?
+const TYPING_DELAY = 800;
```

Comment:
This would be the third place this line shows up in the app. I can make a follow up PR to extract the constant to keep this consistent.

Reply:
Please make sure all occurrences have similar effects on search, to avoid subtle problems if single shared ever changes in the future.

Reply:
Hmm good point. Other uses have this to handle _pagination_ requests, but this is for a search filter. Same general goal I suppose (easing number of requests to the server), but maybe I'll keep this separate instead for the reason you mentioned.